src/kernel/scheduling/thread.c

```diff
@@ -39,7 +41,12 @@ thread_t* create_thread(char* thread_name, void (*_entry_point)(void *), void* a
     if (!is_supervisor) {
         // This piece of code is temporary, just to test a userspace task, it run just an infinite loop.
         pretty_logf(Verbose, "vmm_data address: 0x%x", &(parent_task->vmm_data));
-        new_thread->execution_frame->rip = prepare_userspace_function(&(parent_task->vmm_data));
+        if (is_elf ) {
+            pretty_logf(Verbose, "Preparing to launch an ELF. entry_pint = 0x%x", (uint64_t) _entry_point);
+            new_thread->execution_frame->rip = (uint64_t)_entry_point;
+        } else {
+            new_thread->execution_frame->rip = prepare_userspace_function(&(parent_task->vmm_data));
```

Comment:
whats the purpose of doing it this way? I feel like you could have `prepare_userspace_function` called when creating the task for the elf, and then `create_thread` doesnt need to know if the thread is for an elf or not. You can just tell it what address to begin executing and itll use that. It would also simplify things a bit.

Reply:
the prepare_userspace_function is/was a temporary function that was basically copying a program (same for the one that now i'm running from ELF) from a char array  into a userspace memory space  and launch it. 

I think this will be removed soon (but i think i'll keep it around until i fix the building of the elf that match the kernel PAGE_SIZE)

Reply:
But yeah, that maybe can be repurposed to prepare the memory space for a new  thread being launched from an executable file.